syntax = "proto3";

package raftKVRpcProctoc;

option cc_generic_services = true;  //开启stub服务

message GetArgs{
    bytes Key = 1;
    bytes ClientId = 2;
    int32 RequestId = 3;
}

message GetReply{
    bytes Err = 1;
    bytes Value = 2;
}

message PutAppendArgs{
    bytes Key = 1;
    bytes Value = 2;
    bytes Op = 3;
    bytes ClientId = 4;
    int32 RequestId = 5;
}

message PutAppendReply{
    bytes Err = 1;
}

// rpc通信
// raft都是调用方，kvserver是服务端
service KvServerRpc
{
    rpc PutAppend(PutAppendArgs) returns(PutAppendReply);
    rpc Get (GetArgs) returns (GetReply);
}